Floofbot/Modules/ModMail.cs

```diff
@@ -0,0 +1,195 @@
+    [Group("modmailconfig")]
+    public class ModMailConfigModule : InteractiveBase
+    {
+        private FloofDataContext _floofDB;
+
+        public ModMailConfigModule(FloofDataContext floofDB)
+        {
+            _floofDB = floofDB;
+        }
+        private Discord.Color GenerateColor()
+        {
+            return new Discord.Color((uint)new Random().Next(0x1000000));
+        }
+
+        private void CheckServerEntryExists(ulong server)
+        {
+            // checks if server exists in database and adds if not
+            var serverConfig = _floofDB.ModMails.Find(server);
+            if (serverConfig == null)
+            {
+                _floofDB.Add(new ModMail
+                {
+                    ServerId = server,
+                    ChannelId = null,
+                    IsEnabled = false,
+                    ModRoleId = null
+                });
+                _floofDB.SaveChanges();
+            }
+        }
+
+        [Command("channel")]
+        [Summary("Sets the channel for the modmail notifications")]
+        public async Task Channel([Summary("Channel (eg #alerts)")]Discord.IChannel channel)
+        {
+            CheckServerEntryExists(Context.Guild.Id);
+            var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+            ServerConfig.ChannelId = channel.Id;
+            _floofDB.SaveChanges();
+            await Context.Channel.SendMessageAsync("Channel updated! I will send modmails to <#" + channel.Id + ">");
+        }
+
+        [Command("toggle")]
+        [Summary("Toggles the modmail module")]
+        public async Task Toggle()
+        {
+
+            // try toggling
+            try
+            {
+                CheckServerEntryExists(Context.Guild.Id);
+                // check the status of logger
+                var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+                if (ServerConfig.ChannelId == null)
+                {
+                    await Context.Channel.SendMessageAsync("Channel not set! Please set the channel before toggling the ModMail feature.");
+                    return;
+                }
+                ServerConfig.IsEnabled = !ServerConfig.IsEnabled;
+                _floofDB.SaveChanges();
+                await Context.Channel.SendMessageAsync("Modmail " + (ServerConfig.IsEnabled ? "Enabled!" : "Disabled!"));
+            }
+            catch (Exception ex)
+            {
+                await Context.Channel.SendMessageAsync("An error occured: " + ex.Message);
+                Log.Error("Error when trying to toggle the modmail: " + ex);
+                return;
+            }
+        }
+        [Command("modrole")]
+        [Summary("OPTIONAL: A Role to Ping When ModMail is Received.")]
+        public async Task SetModRole(string roleName = null)
+        {
+            var ServerConfig = _floofDB.ModMails.Find(Context.Guild.Id);
+            if (roleName == null)
```

Comment:
Invalid argument check should happen first

Reply:
done